### DISCUSSION TITLE
CICD with Github Actions (workflow)

### DISCUSSION THREAD

**Discussion by ofirTelecareBill** [serverless/serverless#11303] [Q&A]
Trying to implement CICD with my serverless project, but it keeps complaining about variable resolution. 

_Cannot resolve serverless.yml: Variables resolution errored with:
  - Cannot resolve variable at "provider.environment.MESSAGE": Value not found at "self" source_
  
This setup works when I do the "serverless deploy" manually, but in Github using the deploy-develop.yml (attached as well) it complanins about the variables. What am I doing wrong?

[serverless - Copy.txt](https://github.com/serverless/serverless/files/9223469/serverless.-.Copy.txt)
[deploy-develop - Copy.txt](https://github.com/serverless/serverless/files/9223474/deploy-develop.-.Copy.txt)

**Comment by pgrzesik** [comment]
Hey @ofirTelecareBill, what version of Serverless are you running locally?

**Reply by ofirTelecareBill** [reply]
Any ideas?

**Reply by ofirTelecareBill** [reply]
@pgrzesik I'm kind of stuck with this .. anything I could do to try to resolve?

**Comment by ofirTelecareBill** [comment]
Hi ... Here you go: (Running Windows 10)

C:\Users\OFIR>serverless --version
Framework Core: 3.11.0
Plugin: 6.2.1
SDK: 4.3.2

**Comment by pgrzesik** [comment]
Hey @ofirTelecareBill, I'm sorry but I don't think I will be able to dive deeper into your issue to help more. One thing that I might suggest is using `${sls:stage}` when trying to resolve stage instead of using some custom variables. Other thing that I might suggest for your specific pattern is the use of stage parameters - https://www.serverless.com/framework/docs/guides/parameters#stage-parameters

**Comment by ofirTelecareBill** [comment]
Thanks, I'll check it out.
